packages/block-library/src/site-logo/edit.js

```diff
@@ -480,21 +482,25 @@ export default function LogoEdit( {
 
 		if ( ! media.id && media.url ) {
 			// This is a temporary blob image.
+			setTemporaryURL( media?.url );
 			setLogo( undefined );
 			return;
 		}
 
+		setTemporaryURL( media?.url );
```

Comment:
What's the reason for setting a temp URL here? The blob URL check above returns early, so the image will be uploaded when this line is reached.

This line causes the Site Logo to be displayed in the upload state when selecting an already uploaded image from the Media Library.

Reply:
Good catch, Must have added to test. forgot to remove it after.